syntax = "proto3";

package common;

option go_package = "github.com/spiceai/spiceai/pkg/proto/common_pb";

message Observation {
  int64 time = 1;
  map<string, double> measurements = 2;
  map<string, string> categories = 3;
  repeated string tags = 4;
}

message Interpretation {
    int64 start = 1;
    int64 end = 2;
    string name = 3;
    repeated string actions = 4;
    repeated string tags = 5;
}

message InterpretationIndices {
  repeated uint32 indicies = 1 [packed=true];
}

message IndexedInterpretations {
  repeated Interpretation interpretations = 1;
  map<int64, InterpretationIndices> index = 2;
}